[quant][eagermode][fix] Fix quantization for DeQuantStub

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49428 [quant][eagermode][fix] Fix quantization for DeQuantStub**

Summary:
Previously dequantstub will be swapped with nn.quantized.DeQuantize regardless of qconfig
reason is we skipped attaching qconfig for DeQuantStub to avoid adding fake quantize module to it
but the correct fix is to skip it in insert observers, this PR fixes the issue.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25569991](https://our.internmc.facebook.com/intern/diff/D25569991)

jerryzh168 committed:
[quant][eagermode][fix] Fix quantization for DeQuantStub

Summary:
Previously dequantstub will be swapped with nn.quantized.DeQuantize regardless of qconfig
reason is we skipped attaching qconfig for DeQuantStub to avoid adding fake quantize module to it
but the correct fix is to skip it in insert observers, this PR fixes the issue.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on test/quantization/test_quantize.py:
```diff
@@ -536,6 +536,24 @@ def test_quantized_embedding(self):
+        self.assertEqual(type(m.quant), nnq.Quantize)
+        self.assertEqual(type(m.dequant), nnq.DeQuantize)
+
+        # check DeQuantStub is not swapped when it doesn't have a qconfig
+        m = QuantStubModel().eval()
+        m.dequant.qconfig = None
+        prepare(m, inplace=True)
+        self.checkObservers(m)
+        convert(m, inplace=True)
+        self.assertEqual(type(m.quant), nnq.Quantize)
+        self.assertEqual(type(m.dequant), DeQuantStub)
```
optional nit: not shadowing `m` (i.e. use `m2` or something similar) would reduce the risk of someone changing the code incorrectly in the future

jerryzh168 replied:
sure

jerryzh168 committed:
Update on "[quant][eagermode][fix] Fix quantization for DeQuantStub"


Summary:
Previously dequantstub will be swapped with nn.quantized.DeQuantize regardless of qconfig
reason is we skipped attaching qconfig for DeQuantStub to avoid adding fake quantize module to it
but the correct fix is to skip it in insert observers, this PR fixes the issue.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25569991](https://our.internmc.facebook.com/intern/diff/D25569991)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][fix] Fix quantization for DeQuantStub"


Summary:
Previously dequantstub will be swapped with nn.quantized.DeQuantize regardless of qconfig
reason is we skipped attaching qconfig for DeQuantStub to avoid adding fake quantize module to it
but the correct fix is to skip it in insert observers, this PR fixes the issue.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25569991](https://our.internmc.facebook.com/intern/diff/D25569991)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][fix] Fix quantization for DeQuantStub"


Summary:
Previously dequantstub will be swapped with nn.quantized.DeQuantize regardless of qconfig
reason is we skipped attaching qconfig for DeQuantStub to avoid adding fake quantize module to it
but the correct fix is to skip it in insert observers, this PR fixes the issue.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25569991](https://our.internmc.facebook.com/intern/diff/D25569991)

[ghstack-poisoned]
